src/gsy_e/gsy_e_core/market_utils.py

```diff
@@ -0,0 +1,76 @@
+"""
+Has to stay in this package because of otherwise circualr imports
+"""
+from gsy_framework.constants_limits import ConstSettings
+from gsy_framework.enums import BidOfferMatchAlgoEnum
+from pendulum import DateTime
+
+
+class MarketCounter:
+    """Base class for market counters"""
+
+    def __init__(self, clearing_interval: int):
+        self._last_time_dispatched = None
+        self.clearing_interval = clearing_interval
+
+    def is_time_for_clearing(self, current_time: DateTime) -> bool:
+        """Return if it is time for clearing according to self.clearing_interval."""
+        if not self._last_time_dispatched:
+            self._last_time_dispatched = current_time
+            return True
```

Comment:
I think that even returning `False` wouldn't make it fail-safe. The main problem here is that this class doesn't know when this method will be called for the first time.

In theory, we could initialize the `FutureMarketCounter` (the only counter we use) at the beginning of the simulation and immediately set `self._last_time_dispatched = <simulation-start-time>`, so that all subsequent calls will have the initial start time as reference. 
The problem is that to do this, the counter should be initialized when the simulation is started. At the moment, instead, `FutureMarketCounter` is a singleton. I don't know if there's a specific reason for this, maybe @Hassan754 can give some more details.

If you feel like this conversation is dragging along, feel free to proceed anyway.

Reply:
I believe there is no strong reason for keeping the `FutureMarketCounter`, at least for now, unless the myco external matcher needs to access it at some point (With a quick search in the changes of this Pr I didn't find a need for it, but I will review thoroughly )

Reply:
Just a suggestion, but couldn't the myco get access to the simulation directly? That way it would have access to everything it needs:
```python
bid_offer_matcher.activate(simulation=self)
# And then you could have (just as an example)
bid_offer_matcher.simulation.future_markets_counter
```
This would probably remove the need for the singleton.
Anyway: this conversation became longer than 4 replies. Feel free to move it to slack if needed, or to drop it completely.

Reply:
Your suggestion works if we only want to use the internal myco. 
I will not post this on slack, because we currently have lots of (more pressing) discussion going on. 
I will open a ticket to deal wit this maybe in the frame of onboarding. Thanks!

Reply:
Ah, but even for teh external, the counter could live in the simulation AFAIS, we only have to expose the "is time for clearing" info via redis.

Reply:
Anyway, lets deal with this in here: https://gridsingularity.atlassian.net/browse/GSYE-322